File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
//declaring the needed variables
    int coins = 0;
    float change;

//prompt for positive value
    do
    {
        change = get_float("Change owed: \n");
    }
    while (change < 0);
//make the input a round number
    change = change * 100;
    change = round(change);
//checks how many 25 cent coins
    while (change >= 25)
    {
        change = change - 25;
        coins++;
    }
// checks how many 10 cent coins
    while (change >= 10)
    {
        change = change - 10;
        coins++;
    }
// checks how many 6 cent coins
    while (change >= 5)
    {
        change = change - 5;
        coins++;
    }
// checks how many 1 cent coins
    while (change >= 1)
    {
        change = change - 1;
        coins++;
    }
//prints the total amount of coins
    printf("%d\n", coins);
}

File: pset6/cash/cash.py
import cs50

# Declaring variables
coins = 0
change = -1

# Promp user for positive value
while change < 0:
    change = cs50.get_float("Change owed: ")
# Make the input round number
change *= 100
change = round(change)
# Check how many 25 cents coins
while change >= 25:
    change -= 25
    coins += 1
# Check how many 10 cents coins
while change >= 10:
    change -= 10
    coins += 1
# Check how many 5 cents coins
while change >= 5:
    change -= 5
    coins += 1
# Check how many 1 cent coins
while change >= 1:
    change -= 1
    coins += 1
# Prints the total amount of coins
print(coins)
